src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java

```diff
@@ -110,6 +125,18 @@ public void auditStarted(AuditEvent event) {
 
     @Override
     public void auditFinished(AuditEvent event) {
+        fileMessages.forEach((fileName, messages) -> {
+            writeFileOpeningTag(fileName);
```

Comment:
@sabaka I tried to do so, however, it leads to test failures, e.g https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/XMLLoggerTest.java#L136

I'm not sure, if this is a correct behavior. As for me, we should force closing tag here.

@rnveach @romani What do you think?